### ISSUE TITLE
Duplicating last line in script creates weird undo history

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#21811]
**Steps to reproduce:**

0. (optionally) Create new script for best testing
1. Write any line in your script. It MUST be the last line of the script
2. Press CTRL + B (default) on that line
3. It will duplicate the line and insert an unexpected newline at the end
4. Press CTRL + Z to undo

Expected: It should remove the duplicated line, so you return to the single line you wrote

What actually happens: the line is removed, but there's an empty line at the end, which wasn't there before undo. What's more interesting, after you CTRL+Z again, the one line you wrote is removed (it is somewhat correct), but then! you can CTRL+Z again, which will restore the line (wtf). Another undo will return your script to the actual state it had before duplication.

So to sum up: if you duplicate line (CTRL+B) when it's last line of the file, instead of one action in undo history, it will create 3, with weird intermediate steps that didn't happen (unless internally ofc). It also appends a \n out of nowhere.

Doing CTRL+B on non-last line produces correct single history entry, but the caret doesn't return to initial position after undo. It Stays on the new line (that could count as separate issue, but might be too trivial to create one).

**Comment by vnen** [comment]
Godot version?
